hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/capacity/AbstractCSQueue.java

```diff
@@ -1729,7 +1731,8 @@ public void updateLastSubmittedTimeStamp() {
     }
   }
 
-  public long getLastSubmittedTimestamp() {
+  @VisibleForTesting
+  long getLastSubmittedTimestamp() {
```

Comment:
As @TaoYang526 also pointed out in the jira, this is also being used by production code so the VisibleForTesting annotation is not applicable for this one.

Reply:
As a second thought: getLastSubmittedTimestamp is called from org.apache.hadoop.yarn.server.resourcemanager.scheduler.capacity.AbstractCSQueue#isInactiveDynamicQueue so technically it could have been private if it wasn't accessed from test code so I think VisibleForTesting is correct here.